examples/sphere.py

```diff
@@ -259,28 +311,50 @@ def sphere_main(algorithm,
     if not outdir.is_dir():
         outdir.mkdir()
 
+    if algorithm in ["cma_mega", "cma_mega_adam"]:
```

Comment:
We should set `dim=None` and `itrs=None` and then have the default behavior depend on what the algorithm is.